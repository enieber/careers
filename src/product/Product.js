import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

class Product extends React.PureComponent {
  render() {
    const {
      img,
      title,
      value,
      description,
      buy,
    } = this.props;

    return (
      <Card
        style={{
          width: '30vw',
          margin: '5vw',
          display: 'flex',
          flexDirection: 'column',
        }}
      >
        <CardMedia
          component="img"
          height="140"
          image={img}
          title="Contemplative Reptile"
        />
        <CardContent
          style={{
            padding: '2vw',
          }}
        >
          <Typography gutterBottom variant="headline" component="h3">
            {title}
          </Typography>
          <Typography gutterBottom variant="headline" component="h4">
            {value}
          </Typography>
          <Typography component="p">
            {description}
          </Typography>
          <Button
            id="button"
            variant="contained"
            color="primary"
            type="button"
            onClick={buy}
          >
           Comprar
          </Button>
        </CardContent>
      </Card>
    );
  }
}

Product.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  buy: PropTypes.func.isRequired,
};

export default Product;
