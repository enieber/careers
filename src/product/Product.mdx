---
name: Product
---

import Product from './Product';
import { Playground, PropsTable } from 'docz';

# RamalItem

## Basic usage

<Playground>
 <div
   style={{display: 'flex'}}
 >
 <Product 
  title={'Product 1'}
  value={'100,00'}
  description={'test description'}
  category={'Category 1'}
  buy={() => {
    alert('functions to buy product');
  }}
/>
</div>
</Playground>



## Properties
<PropsTable of={Product} />

