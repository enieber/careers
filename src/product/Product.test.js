import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import Product from './Product';

describe('Testing Product Component', () => {
  it('shoult renders without crashing', () => {
    const wrapper = shallow(
      <Product
        img=""
        title="Product 1"
        value="100,00"
        description="test description"
        category="Category 1"
        buy={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    const propsChange = {
      img: '',
      title: 'test 1',
      value: '10,00',
      description: 'description',
      category: 'category test',
      buy: () => {},
    };
    wrapper.setProps(propsChange);
    expect(wrapper.instance().props).toEqual(propsChange);
  });
  it('simulates buy click', () => {
    const onButtonClick = sinon.spy();
    const wrapper = shallow(
      <Product
        img=""
        title="Product 1"
        value="100,00"
        description="test description"
        category="Category 1"
        buy={onButtonClick}
      />,
    );
    wrapper.find('#button').simulate('click');
    expect(onButtonClick.callCount).toEqual(1);
  });
});
