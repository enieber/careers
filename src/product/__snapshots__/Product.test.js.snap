// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Product Component shoult renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product
    buy={[Function]}
    category="Category 1"
    description="test description"
    img=""
    title="Product 1"
    value="100,00"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardMedia)
          component="img"
          height="140"
          image=""
          title="Contemplative Reptile"
        />,
        <WithStyles(CardContent)
          style={
            Object {
              "padding": "2vw",
            }
          }
        >
          <WithStyles(Typography)
            component="h3"
            gutterBottom={true}
            variant="headline"
          >
            Product 1
          </WithStyles(Typography)>
          <WithStyles(Typography)
            component="h4"
            gutterBottom={true}
            variant="headline"
          >
            100,00
          </WithStyles(Typography)>
          <WithStyles(Typography)
            component="p"
          >
            test description
          </WithStyles(Typography)>
          <WithStyles(Button)
            color="primary"
            id="button"
            onClick={[Function]}
            type="button"
            variant="contained"
          >
            Comprar
          </WithStyles(Button)>
        </WithStyles(CardContent)>,
      ],
      "style": Object {
        "display": "flex",
        "flexDirection": "column",
        "margin": "5vw",
        "width": "30vw",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": "img",
          "height": "140",
          "image": "",
          "title": "Contemplative Reptile",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              component="h3"
              gutterBottom={true}
              variant="headline"
            >
              Product 1
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              component="h4"
              gutterBottom={true}
              variant="headline"
            >
              100,00
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              component="p"
            >
              test description
            </WithStyles(Typography)>,
            <WithStyles(Button)
              color="primary"
              id="button"
              onClick={[Function]}
              type="button"
              variant="contained"
            >
              Comprar
            </WithStyles(Button)>,
          ],
          "style": Object {
            "padding": "2vw",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Product 1",
              "component": "h3",
              "gutterBottom": true,
              "variant": "headline",
            },
            "ref": null,
            "rendered": "Product 1",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "100,00",
              "component": "h4",
              "gutterBottom": true,
              "variant": "headline",
            },
            "ref": null,
            "rendered": "100,00",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "test description",
              "component": "p",
            },
            "ref": null,
            "rendered": "test description",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Comprar",
              "color": "primary",
              "id": "button",
              "onClick": [Function],
              "type": "button",
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Comprar",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardMedia)
            component="img"
            height="140"
            image=""
            title="Contemplative Reptile"
          />,
          <WithStyles(CardContent)
            style={
              Object {
                "padding": "2vw",
              }
            }
          >
            <WithStyles(Typography)
              component="h3"
              gutterBottom={true}
              variant="headline"
            >
              Product 1
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="h4"
              gutterBottom={true}
              variant="headline"
            >
              100,00
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="p"
            >
              test description
            </WithStyles(Typography)>
            <WithStyles(Button)
              color="primary"
              id="button"
              onClick={[Function]}
              type="button"
              variant="contained"
            >
              Comprar
            </WithStyles(Button)>
          </WithStyles(CardContent)>,
        ],
        "style": Object {
          "display": "flex",
          "flexDirection": "column",
          "margin": "5vw",
          "width": "30vw",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": "img",
            "height": "140",
            "image": "",
            "title": "Contemplative Reptile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                component="h3"
                gutterBottom={true}
                variant="headline"
              >
                Product 1
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                component="h4"
                gutterBottom={true}
                variant="headline"
              >
                100,00
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                component="p"
              >
                test description
              </WithStyles(Typography)>,
              <WithStyles(Button)
                color="primary"
                id="button"
                onClick={[Function]}
                type="button"
                variant="contained"
              >
                Comprar
              </WithStyles(Button)>,
            ],
            "style": Object {
              "padding": "2vw",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Product 1",
                "component": "h3",
                "gutterBottom": true,
                "variant": "headline",
              },
              "ref": null,
              "rendered": "Product 1",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "100,00",
                "component": "h4",
                "gutterBottom": true,
                "variant": "headline",
              },
              "ref": null,
              "rendered": "100,00",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "test description",
                "component": "p",
              },
              "ref": null,
              "rendered": "test description",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Comprar",
                "color": "primary",
                "id": "button",
                "onClick": [Function],
                "type": "button",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Comprar",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
